pkg/cloudprovider/drift.go

```diff
@@ -73,3 +84,33 @@ func (c *CloudProvider) isAMIDrifted(ctx context.Context, machine *v1alpha5.Mach
 	}
 	return !lo.Contains(lo.Keys(mappedAMIs), instance.ImageID), nil
 }
+
+// Checks if the security groups are drifted, by comparing the AWSNodeTemplate.Status.SecurityGroups
+// to the ec2 instance security groups
+func (c *CloudProvider) areSecurityGroupsDrifted(ec2Instance *instance.Instance, nodeTemplate *v1alpha1.AWSNodeTemplate) (bool, error) {
+	// nodeTemplate.Spec.SecurityGroupSelector can be nil if the user is using a launchTemplateName to define SecurityGroups
+	// Karpenter will not drift on changes to securitygroup in the launchTemplateName
+	if nodeTemplate.Spec.LaunchTemplateName != nil {
```

Comment:
I was thinking about this more and I'm wondering if we should encapsulate this behavior in the AWSNodeTemplate controller. Meaning, that we should set the resolved security groups to `nil` when a LaunchTemplateName is defined so that we can just evaluate using L98 and we don't have think about this specific case in this context as well

Reply:
Currently, AWSNodeTemplate controller does place a `nil` for the status when LaunchTemplateName is set. We should't have errors, if the user use LaunchTemplateName. In L98 we will error if the status does not have any SGs